implement-shell-tools/cat/cat.py

```diff
@@ -0,0 +1,40 @@
+import argparse
+
+parser = argparse.ArgumentParser(
+    prog="cat.py",
+    description="Display contents of files with optional line numbering."
+)
+
+parser.add_argument("-n", action="store_true", help="Number all lines")
+parser.add_argument("-b", action="store_true", help="Number non-blank lines")
+parser.add_argument("files", nargs="+", help="Files to display")
+
+args = parser.parse_args()
+
+# Check if user used -n or -b
+number_all_lines = args.n
+number_non_blank = args.b
+
+# Loop through all file paths
+for file_path in args.files:
+    with open(file_path, "r", encoding="utf-8") as file:
```

Comment:
Here and in another scripts, you could check that the file actually exists and print a user friendly message if it doesn't.